doc/api/n-api.md

```diff
@@ -118,6 +118,96 @@ must make use exclusively of N-API by restricting itself to using
 and by checking, for all external libraries that it uses, that the external
 library makes ABI stability guarantees similar to N-API.
 
+## Building
+
+Unlike modules written in JavaScript, developing and deploying Node.js native
+addons using N-API requires an additional set of tools. Besides the basic
+tools required to develop for Node.js, the native addon developer requires a
+toolchain that can compile C and C++ code into a binary. In addition,
+depending upon how the native addon is deployed, the *user* of the native
+addon will also need to have a C/C++ toolchain installed.
+
+For Linux developers, the necessary C/C++ toolchain packages are readily
+available. For many developers, the [LLVM](https://llvm.org) compiler
```

Comment:
I'm wondering if we should reference GCC instead since I think that is what we use in the community build/testing on many platforms.

Reply:
I've added GCC as the primary recommendation in addition to LLVM as an alternative.